src/dft/backends/rocfft/backward.cpp

```diff
@@ -52,21 +56,33 @@ rocfft_execution_info get_bwd_info(dft::detail::commit_impl<prec, dom> *commit)
 template <typename descriptor_type>
 ONEMKL_EXPORT void compute_backward(descriptor_type &desc,
                                     sycl::buffer<fwd<descriptor_type>, 1> &inout) {
+    const std::string func_name = "compute_backward(desc, inout)";
     detail::expect_config<dft::config_param::PLACEMENT, dft::config_value::INPLACE>(
         desc, "Unexpected value for placement");
     auto commit = detail::checked_get_commit(desc);
     auto queue = commit->get_queue();
     auto plan = detail::get_bwd_plan(commit);
     auto info = detail::get_bwd_info(commit);
+    auto offsets = detail::get_offsets(commit);
+
+    if constexpr (std::is_floating_point_v<fwd<descriptor_type>>) {
+        offsets[0] *= 2; // offset is supplied in complex but we offset scalar pointer
+    }
+    if (offsets[0] != offsets[1]) {
```

Comment:
> Can you check this at commit time by looking at the placement commit parameter to see if its needed?

I could, but it is easier to do here when I also know direction of the transform. At commit time I would need to check both directions to see if either makes sense.

> Also, why is this a requirement when you are just applying the offset to the pointer yourself? rocFFT doesn't ever see the offset.

For in-place transforms rocfft accepts single pointer, so we can only apply one offset. That only makes sense if both offsets are the same.

Reply:
> For in-place transforms rocfft accepts single pointer, so we can only apply one offset. That only makes sense if both offsets are the same.

Ok that makes sense.


wrt checking at commit time, I'm speaking specifically about the `offset[0] != offset[1]` test, I believe that is orthogonal to direction.

Reply:
It is not. For real transforms direction determines which offset is specified in reals and which in complex values. This is the problem of having input/output values instead of forward/backward ...

Reply:
Ok, I see. Thank you for explaining.